codegen/src/api/mod.rs

```diff
@@ -357,7 +389,16 @@ impl RuntimeGenerator {
                     T: ::subxt::Config,
                 {
                     #(
-                        pub fn #pallets_with_storage(&self) -> #pallets_with_storage::storage::StorageApi<'a, T> {
+                        pub fn #pallets_with_storage(&self) -> Result<#pallets_with_storage::storage::StorageApi<'a, T>, ::subxt::MetadataError> {
```

Comment:
Cool, looks good!

Do you think we should add this same `_unchecked` thing to the `ConstantsApi` too, since, as of Andrew Jones's change, the constant values are fetched from the downloaded-at-runtime metadata and then decoded based on the statically generated code, so as with transactions and storage, a difference in metadata could silently break or mess with decoding of the constants?